### ISSUE TITLE
Check for CVEs

### ISSUE THREAD

**Issue by kitarp29** [kitarp29/kube-ez#46]
Right now the CI is set up such that every successful build of Docker Image will be pushed to ghcr.
But this is very risky, as I am not scanning for CVEs. Neither I have Image scanning open on my Docker Hub as I am broke.
So we need to find some CLI-based Docker Image Scanning Jobs in the CI.
Refer to This: [Here](https://10clouds.com/blog/devops/docker-image-security-a-handy-comparison-of-the-top-tools/)
![image](https://github.com/kitarp29/kube-ez/assets/56778773/cc0fb0ec-4ab8-473d-b5f2-808b3e9fe94a)

**Comment by kitarp29** [comment]
Look into this
https://github.com/marketplace/actions/docker-scout

**Comment by kitarp29** [comment]
So I have enabled Docker Scout on my repo.
It seems to be free in the early access version. ( Let's see when it is free xD)
Anyway, this is the changes I saw after fixing it:

All I had to do is to upgrade the base image. This means I just had to build an image so that it gets the latest of the base image.
It raises a good itch in my head, I should have a CRON job for this. It runs once a month or so...

I know my CI will build and test, do functionality won't break!
Only demerit I see is the documentation will be outdated with the tag each time. And I really don't want to make any commit with a CI runner!
![image](https://github.com/kitarp29/kube-ez/assets/56778773/d4639ebe-2cc7-45ec-b33e-b1b970b7a2e2)

**Comment by kitarp29** [comment]
Anyway, this only fixes the CVEs on the image layer.
Let's plan and build something for the code level.
![image](https://github.com/kitarp29/kube-ez/assets/56778773/a43ab966-a332-47ea-be08-50695f21a0a7)

**Comment by kitarp29** [comment]
Ok, so I set up Snyk to scan my codebase.
Now this got really interesting!
Most of the CVEs are dependency-related or Kubernetes-YAML related.
It is interesting because it did not say any part of the code has a CVE 🤯

Either one of the two things is happening here:
 - Snyk on the free level is bullcrap and not working as expected. I don't even see suggestions to fix my CVE on the dashboard.
 - I am the proof of singularity and I write code with no CVE 😂
 
Anyway my search for a Code scanning tool for my CI is not done yet then!
I mean, the repo has CodeQL setup but you know it's not the same feeling.
I learned some new things today for sure.
